### PR TITLE
Enumerable extensions

### PR DESCRIPTION
**Pull request by yuriyoleynikov**
for @Igorbek

### CODE REVIEW

**Comment by yuriyoleynikov** [comment]
@Igorbek check this and merge

File: tests/YoTennis.Tests/Test/EnumerableExtensionsTests.cs [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+            var forFirstCompare = new FirstEqualityComparer();
+            var sequence = new[] { "one", "two", "three" };
+            var result = sequence.WithoutElement("five", forFirstCompare);
+            result.Should().BeEquivalentTo(new[] { "one", "two", "three" });
+        }
+
+        [Fact]
+        public void Test_WithoutElement_Int_WithCompare3()
+        {
+            var IEComparer = new IntEqualityComparer();
```
Comment (Igorbek) [inline tests/YoTennis.Tests/Test/EnumerableExtensionsTests.cs]: do not call local variables like this. local variables must be `camelCased`. `PascalCase` is used for classes, properties and so on. Moreover, format `IPascalCase` is meant to be used for interfaces. In your case just use `comparer` here.

File: tests/YoTennis.Tests/Test/EnumerableExtensionsTests.cs [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+        [Fact]
+        public void Test_WithoutElement_Int_WithCompare5()
+        {
+            var IEComparer2 = new IntDigitEqualityComparer();
+            var sequence = new[] { 1, 11, 111, 1111 };
+            var result = sequence.WithoutElement(88888, IEComparer2);
+            result.Should().BeEquivalentTo(new[] { 1, 11, 111, 1111 });
+        }
+
+        class FirstEqualityComparer : IEqualityComparer<string>
```
Comment (Igorbek) [inline tests/YoTennis.Tests/Test/EnumerableExtensionsTests.cs]: `First`**`Char`**`EqualityComparer`

File: tests/YoTennis.Tests/Test/EnumerableExtensionsTests.cs [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        class IntDigitEqualityComparer : IEqualityComparer<int>
+        {
+            public bool Equals(int x, int y)
+            {
+                return x.ToString().Length == y.ToString().Length;
```
Comment (Igorbek) [inline tests/YoTennis.Tests/Test/EnumerableExtensionsTests.cs]: :)

**Review by Igorbek — CHANGES_REQUESTED** [review]

**Pull request merged**